README.md

```diff
@@ -0,0 +1,181 @@
+
+```
+gpu_list:
+    - 0
+    - 1
+dump_path: './outputs/'
+defaults:
+    - data: iwcd_postveto_nomichel
+    - data/dataset: iwcd_cnn
+    - model: classifier
+    - model/feature_extractor: resnet18
+    - model/classification_network: resnet_fc
+    - engine: classifier
+    - tasks/train: train_resnet
+    - optimizers@tasks.train.optimizers: adam
+    - sampler@tasks.train.data_loaders.train.sampler: subset_random
+    - sampler@tasks.train.data_loaders.validation.sampler: subset_sequential
+    - tasks/evaluate: test
+    - sampler@tasks.evaluate.data_loaders.test.sampler: subset_sequential
+
+```
+
+These subconfigs correspond to config groups (more on config groups can be found in the hydra documentation). Generally, all the parameters related to a certain task or object are grouped into separate config groups. The defaults list in the main config contains default config for relevant config groups.
+
+Some of the config groups listed are uised to instantiate objects using hydra. For example `subset_random.yaml` contains:
+
+```
+# @package _group_._name_
+_target_: torch.utils.data.sampler.SubsetRandomSampler
+```
+
+And can be used to instantiate a SubsetRandomSampler by:
+
+```
+from hydra.utils import instantiate
+
+sampler = instantiate(sampler_config)
+```
+
+Other config packages are used for calling functions. The config object is passed as a dict, and then its parameters are used as part of a function. In particular, this is how tasks like training and testing are organized. For example the training task config `train_resnet.yaml` contains parameters used to set up the training run:
+
+```
+# @package _group_
+epochs: 20
+
+report_interval: 10
+val_interval: 10
+num_val_batches: 32
+
+checkpointing: False
+
+data_loaders:
+  train:
+    split_key: train_idxs
+    batch_size: 512
+    num_workers: 4
+    transforms:
+      - horizontal_flip
+      - vertical_flip
+  validation:
+    split_key: val_idxs
+    batch_size: 512
+    num_workers: 4
+```
+Which is then used in the call to the task:
+
+```
+def train(self, train_config):
+    ...
+    # initialize training params
+    epochs          = train_config.epochs
+    report_interval = train_config.report_interval
+    val_interval    = train_config.val_interval
+    num_val_batches = train_config.num_val_batches
+    checkpointing   = train_config.checkpointing
+```
+
+Config parameters and groups can be overridden in the command line. For example, to run using default configuration:
+
+```
+python main.py
+```
+
+And to override the list of GPUs to use:
+
+```
+python main.py gpu_list=[0]
+```
+
+Default config groups (but not parameters) can also be overridden in the defaults list:
+
+```
+    - sampler@tasks.train.data_loaders.train.sampler: subset_random
+```
+
+For additional help with comands, Hydra comes with documentation that can be accessed in the command line using:
+
+```
+--hydra-help
+```
```

Comment:
Might be worth also moving this to the top where the readme also links to the Hydra website for documentation